Adds support for Twitter cards and OpenGraph metadata, and inlines Google Fonts

konklone opened the pull request:
This pull request adds support for Twitter cards, OpenGraph metadata for sharing in other platforms, `<link rel="canonical">` links, and follows the [18f.gsa.gov standard method](https://github.com/18F/18f.gsa.gov/pull/672) of inlining Google Fonts to avoid third party service requests.

Twitter/OG metadata and `<link rel="canonical">` links will only be enabled if the individual site's `_config.yml` adds a `url` field (which will be used to populate the `canonical` link). If other fields don't exist, nothing will break, but the fields will be empty, so adding `title` and `description` fields are advised.

A site can add site-level metadata in `_config.yml`, like so:

```yaml
title: Joining 18F
url: https://pages.18f.gov/joining-18f
description: Information on open positions at 18F, and how to join.
```

And any page can override the site-level `title` and `description` in its YAML front-matter.

Again, this metadata will only be used if a site adds a `url` parameter to its `_config.yml`. This is a backwards-compatible update, so existing sites which don't do that will be unaffected.

konklone commented:
For an example of an update to an 18F Pages site to use this behavior, see https://github.com/18F/joining-18f/pull/293. cc @afeld

afeld commented on lib/guides_style_18f/includes/header.html:
```diff
@@ -3,6 +3,51 @@
+
+
+
+{% if site.url %}
```
I'd only fence the tags that depend on site.url...no reason to include the others.

konklone replied:
It's a way to make this backwards-compatible and opt-in. I could add a bunch of fences, but this seemed cleaner. I can do that, though.

afeld commented:
I defer to the front end folks about font inlining, but the meta tags look good to me!

juliaelman commented:
👍  👍  👍  to self hosting the font files, rather than link out! This looks good to me and would be great to see this as a consistent practice on all of our projects.

konklone commented:
@afeld I made each element of the PR conditional on the relevant field, rather than having all them contingent on `site.url`.

I also removed the duplicate `<title>` tag I created when adding the `<title>` from the metadata, and updated the new `<title>` tag to use the same format as the old one.

konklone commented:
I think this is ready for merge, and if possible I'd love a merge (plus a gem deploy) today so that we can benefit from this when promoting our infosec positions this week.

afeld commented:
We're ok to vendor the fonts from a licensing perspective, yeah?

konklone commented:
Yes, Google Fonts are all openly licensed in some form, and Google offers a bulk download of all of them:

https://github.com/google/fonts#download-all-google-fonts

We currently inline Google Fonts on 18f.gsa.gov, and a host of other websites.
